app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala

```diff
@@ -80,44 +79,42 @@ class ReminderController @Inject() (val helpToSaveReminderService: HelpToSaveRem
       withErrors ⇒ {
         Ok(reminderFrequencySet(withErrors))
       },
-
       success ⇒
-        {
-          htsContext.userDetails match {
-
-            case Left(missingUserInfos) ⇒
-              logger.warn(s"Email was verified but missing some user info ${missingUserInfos}")
-
-              internalServerError()
-
-            case Right(userInfo) ⇒
-              val result: EitherT[Future, GetEmailError, HtsUser] = for {
-                emailRetrieved ← helpToSaveService.getConfirmedEmail.leftMap(GetEmailError.HtsEmailError)
-                htsUser ← helpToSaveReminderService.updateHtsUser(HtsUser(Nino(htsContext.nino),
-                                                                          emailRetrieved.getOrElse(""), userInfo.forename, true,
-                                                                          daysToReceive = DateToDaysMapper.d2dMapper.getOrElse(success.reminderFrequency, Seq(1)))) //.leftMap(GetEmailError.UpdateHtsUserError)
-                  .leftMap[GetEmailError](GetEmailError.UpdateHtsUserError)
-
-              } yield htsUser
-
-              result.fold({
-                case GetEmailError.HtsEmailError(e) ⇒
-                  logger.warn(s"Could not get email from hts service: $e")
-                  internalServerError()
-
-                case GetEmailError.UpdateHtsUserError(e) ⇒
-                  logger.warn(s"Could not update the HtsUser details on Hts Reminder service: $e")
+        htsContext.userDetails match {
+          case Left(missingUserInfos) ⇒
+            logger.warn(s"Email was verified but missing some user info ${missingUserInfos}")
+            internalServerError()
+
+          case Right(userInfo) ⇒
+            helpToSaveService.getConfirmedEmail.value.flatMap{
+              _.fold(
+                noEmailError ⇒ {
+                  logger.warn(s"An error occurred while accessing confirmed email service for user: ${userInfo.nino}")
                   internalServerError()
-
-              }, { htsUser ⇒
-                logger.info("Successfully updated htsUser")
-
-                SeeOther(routes.ReminderController.getRendersConfirmPage(crypto.encrypt(htsUser.email), success.reminderFrequency).url)
-              })
-          }
-
-        }
-    )
+                },
+                emailRetrieved ⇒
+                  emailRetrieved match {
+                    case Some(email) ⇒
+                      if (email.length > 0)
```

Comment:
You can do this more tidily using a guard, so say:

      case Some(email) if !email.isEmpty ⇒
 
and later

      case Some(_) =>
                      logger.warn(s"Empty email retrieved for user: ${userInfo.nino}")

Reply:
Great, thanks